api/models.py

```diff
@@ -188,3 +73,17 @@ class User(Base):
     name = db.Column(db.String(200), nullable=False)
     password = db.Column(db.String(200), nullable=False)
     admin = db.Column(db.Boolean, default=False)
+
+
+@event.listens_for(Session, "after_flush")
```

Comment:
also you have your logging on debug level. that's fine for such low level logs. but you still need some info logs at the start of the endpoint hander, because you can't have debug on all the time

Reply:
after_flush called before commit. The data that was in the rollback will not be logged

Reply:
if you do session.commit() that invokes flush as well, yes. however, one might call session.flush() explicitly, on it's own. moreover, by default sqlalchemy flushes on every select. and then rollback may happen for instance in case of exception. in that situation you'll end up with log statements for rolled back operations with no log for rollback itself.